fastly/block_fastly_service_v1_backend.go

```diff
@@ -9,169 +9,19 @@ import (
-			},
-			"request_condition": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "Name of a condition, which if met, will select this backend during a request.",
-			},
-			"shield": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "The POP of the shield designated to reduce inbound load.",
-			},
-			"use_ssl": {
-				Type:        schema.TypeBool,
-				Optional:    true,
-				Default:     false,
-				Description: "Whether or not to use SSL to reach the Backend",
-			},
-			"max_tls_version": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "Maximum allowed TLS version on SSL connections to this backend.",
-			},
-			"min_tls_version": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "Minimum allowed TLS version on SSL connections to this backend.",
-			},
-			"ssl_ciphers": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "Comma sepparated list of ciphers",
-			},
-			"ssl_check_cert": {
-				Type:        schema.TypeBool,
-				Optional:    true,
-				Default:     true,
-				Description: "Be strict on checking SSL certs",
-			},
-			"ssl_hostname": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "SSL certificate hostname",
-				Deprecated:  "Use ssl_cert_hostname and ssl_sni_hostname instead.",
-			},
-			"ssl_ca_cert": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "CA certificate attached to origin.",
-			},
-			"ssl_cert_hostname": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "SSL certificate hostname for cert verification",
-			},
-			"ssl_sni_hostname": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "SSL certificate hostname for SNI verification",
-			},
-			"ssl_client_cert": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "SSL certificate file for client connections to the backend.",
-				Sensitive:   true,
-			},
-			"ssl_client_key": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Default:     "",
-				Description: "SSL key file for client connections to backend.",
-				Sensitive:   true,
-			},
+type BackendServiceAttributeHandler struct {
+	*DefaultServiceAttributeHandler
+}
 
-			"weight": {
-				Type:        schema.TypeInt,
-				Optional:    true,
-				Default:     100,
-				Description: "The portion of traffic to send to a specific origins. Each origin receives weight/total of the traffic.",
-			},
+func NewServiceBackend() ServiceAttributeDefinition {
+	return &BackendServiceAttributeHandler{
+		&DefaultServiceAttributeHandler{
+			key: "backend",
```

Comment:
~I expected to see `schema` set here.~

This is done in the `Register` method.

Reply:
We have to set `key` here to be used below. I agree with setting it here and how it is used below as this will lend itself well for templating the `Register` methods for future additions. Rather than having to update the string in every `Register` function.